# Lab 2

In [57]:
import warnings
warnings.simplefilter('ignore')

import pandas as pd

df = pd.read_csv('s3://ia.340.fall.semester.2025qwerty/Retail_Food_Stores.csv') # load the data from s3
df.head() #show the top 5 rows

County  License Number Operation Type Establishment Type  Entity Name  \
0    ALBANY          750931          Store                 AC  EVANS JULIA   
1     KINGS          619451          Store                 AC  159-MP CORP   
2     KINGS          751355          Store                 AC    YDHNY INC   
3  SARATOGA          722816          Store                 AC     IANS LLC   
4   SUFFOLK          719057          Store                 AC   HO 110 INC   

   DBA Name Street Number Street Name Address Line 2 Address Line 3  \
0   TERROSA             8   VERDA AVE            NaN            NaN   
1  FOODTOWN           159    N 3RD ST            NaN            NaN   
2     YDHNY            25    N 5TH ST            NaN            NaN   
3      IANS          1500        RT 9            NaN            NaN   
4     EXXON          1350      RT 110            NaN            NaN   

           City State  Zip Code  Square Footage  \
0   CLARKSVILLE    NY     12041             NaN   
1      BROOKLYN    NY     11211          9000.0   
2      BROOKLYN    NY     11249             NaN   
3  CLIFTON PARK    NY     12065             NaN   
4   FARMINGDALE    NY     11735          1000.0   

                         Georeference  
0  POINT (-73.949128787 42.577416307)  
1  POINT (-73.959567442 40.715342402)  
2  POINT (-73.963025857 40.719355977)  
3  POINT (-73.742465504 42.837221945)  
4  POINT (-73.423437971 40.737553896)

In [58]:
df.shape #show the number of rows and columns 

(24221, 15)

In [59]:
df.info() # shows the data types and number of null values. 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 24221 entries, 0 to 24220
Data columns (total 15 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   County              24218 non-null  object 
 1   License Number      24221 non-null  int64  
 2   Operation Type      24221 non-null  object 
 3   Establishment Type  24221 non-null  object 
 4   Entity Name         24221 non-null  object 
 5   DBA Name            24221 non-null  object 
 6   Street Number       24160 non-null  object 
 7   Street Name         24221 non-null  object 
 8   Address Line 2      288 non-null    object 
 9   Address Line 3      1 non-null      object 
 10  City                24220 non-null  object 
 11  State               24221 non-null  object 
 12  Zip Code            24221 non-null  int64  
 13  Square Footage      19294 non-null  float64
 14  Georeference        24221 non-null  object 
dtypes: float64(1), int64(2), object(12)
memory usage: 2.8

In [60]:
df.describe() #statistical summary of the data

License Number      Zip Code  Square Footage
count    24221.000000  24221.000000    19294.000000
mean    625829.464679  11856.219851     6742.507515
std     191635.308154   1368.924807    19916.203726
min      10008.000000   6390.000000        0.000000
25%     610637.000000  10958.000000     1000.000000
50%     721708.000000  11379.000000     1800.000000
75%     745316.000000  12801.000000     4000.000000
max     763163.000000  14905.000000   500000.000000

## A. Viewing Data

This dataset has 24,221 rows and 15 columns, including those that are numeric (zip-code, license number) and textual (county, street name, and more). At first impression, I was surprised to see georeference as a column and and the large number of rows.

In [61]:
df = df.rename(columns={"Square Footage": "Area"}) #renaming column
df.head()

County  License Number Operation Type Establishment Type  Entity Name  \
0    ALBANY          750931          Store                 AC  EVANS JULIA   
1     KINGS          619451          Store                 AC  159-MP CORP   
2     KINGS          751355          Store                 AC    YDHNY INC   
3  SARATOGA          722816          Store                 AC     IANS LLC   
4   SUFFOLK          719057          Store                 AC   HO 110 INC   

   DBA Name Street Number Street Name Address Line 2 Address Line 3  \
0   TERROSA             8   VERDA AVE            NaN            NaN   
1  FOODTOWN           159    N 3RD ST            NaN            NaN   
2     YDHNY            25    N 5TH ST            NaN            NaN   
3      IANS          1500        RT 9            NaN            NaN   
4     EXXON          1350      RT 110            NaN            NaN   

           City State  Zip Code    Area                        Georeference  
0   CLARKSVILLE    NY     12041     NaN  POINT (-73.949128787 42.577416307)  
1      BROOKLYN    NY     11211  9000.0  POINT (-73.959567442 40.715342402)  
2      BROOKLYN    NY     11249     NaN  POINT (-73.963025857 40.719355977)  
3  CLIFTON PARK    NY     12065     NaN  POINT (-73.742465504 42.837221945)  
4   FARMINGDALE    NY     11735  1000.0  POINT (-73.423437971 40.737553896)

In [62]:
df.isnull().sum() #show the number of rows containing null value

County                    3
License Number            0
Operation Type            0
Establishment Type        0
Entity Name               0
DBA Name                  0
Street Number            61
Street Name               0
Address Line 2        23933
Address Line 3        24220
City                      1
State                     0
Zip Code                  0
Area                   4927
Georeference              0
dtype: int64

In [65]:
## df = df.dropna() #drop null values
## df.isnull().sum()

### B. Cleaning Data

I renamed the "Square Footage" column to "Area" for sake of simplicity. I found many missing values, particularly within the columns "Address Line 2" which had 23,933 missing values, and "Address Line 3" which had 24,220. I did not handle or drop my missing values because if I did, I would have gotten rid of all my data.

In [63]:
df['City'] #select a column 

0          CLARKSVILLE
1             BROOKLYN
2             BROOKLYN
3         CLIFTON PARK
4          FARMINGDALE
             ...      
24216    RICHMOND HILL
24217         BROOKLYN
24218       BINGHAMTON
24219         WATERLOO
24220         BROOKLYN
Name: City, Length: 24221, dtype: object

In [64]:
df.sort_values(by='Area') #sort data by a column

County  License Number Operation Type Establishment Type  \
7658         MONROE          260413          Store                 AC   
8142          KINGS          612209          Store                 AC   
8127         QUEENS          638547          Store                 AC   
19369  ST. LAWRENCE          400230          Store                  A   
8158     WASHINGTON          530190          Store                 AC   
...             ...             ...            ...                ...   
24187      SARATOGA          719075          Store                  A   
24191         KINGS          756929          Store                 AC   
24199         BRONX          743073          Store                 AC   
24200         BRONX          751044          Store                 AC   
24209         KINGS          734161          Store               ABCH   

                            Entity Name                  DBA Name  \
7658   DELTA SONIC CAR WASH SYSTEMS INC             KISS MART 813   
8142               BED-ROSS GROCERY LLC          BED-ROSS GROCERY   
8127              JAY DELI GROCERY CORP          JAY DELI GROCERY   
19369               DOLGEN NEW YORK LLC       DOLLAR GENERAL 6066   
8158                      SANDORA PENNY   NORTHEAST CRNR HERB FRM   
...                                 ...                       ...   
24187               DOLGEN NEW YORK LLC     DOLLAR GENERAL #15734   
24191                   YOUGH TO GO LLC                     YOUGH   
24199    FOAD MUTHANA & MOHAMAD RASHEED  ARCHER CONVENIENCE STORE   
24200            M&A FRUITS & DELI CORP         M&A FRUITS & DELI   
24209                   KAFF BAKERY INC                  M BAKERY   

      Street Number                  Street Name Address Line 2  \
7658            718                    E MAIN ST            NaN   
8142            104   ROSS ST (END OF RODNEY ST)            NaN   
8127          54-16                 FLUSHING AVE            NaN   
19369           232                    N MAIN ST            NaN   
8158            185                 LAKE NEBO RD            NaN   
...             ...                          ...            ...   
24187           162                  W MILTON RD            NaN   
24191           203  KNICKERBOCKER AVE. STORE LE            NaN   
24199          1501             WHITEPLAINS ROAD            NaN   
24200          1976       ARTHUR AVENUE STORE 76            NaN   
24209          1906                     AVENUE M            NaN   

      Address Line 3          City State  Zip Code  Area  \
7658             NaN     ROCHESTER    NY     14605   0.0   
8142             NaN      BROOKLYN    NY     11249   0.0   
8127             NaN       MASPETH    NY     11378   0.0   
19369            NaN       MASSENA    NY     13662   0.0   
8158             NaN      FORT ANN    NY     12827   0.0   
...              ...           ...   ...       ...   ...   
24187            NaN  BALLSTON SPA    NY     12020   NaN   
24191            NaN      BROOKLYN    NY     11237   NaN   
24199            NaN         BRONX    NY     10462   NaN   
24200            NaN         BRONX    NY     10457   NaN   
24209            NaN      BROOKLYN    NY     11230   NaN   

                             Georeference  
7658   POINT (-77.592955836 43.160587011)  
8142    POINT (-73.961658475 40.70364621)  
8127   POINT (-73.912957606 40.715126526)  
19369  POINT (-74.900179419 44.942878063)  
8158   POINT (-73.560179478 43.442453315)  
...                                   ...  
24187  POINT (-73.928079146 43.036866222)  
24191  POINT (-73.926467306 40.703570941)  
24199  POINT (-73.863768994 40.837949948)  
24200  POINT (-73.892976155 40.846822276)  
24209  POINT (-73.955917626 40.618348234)  

[24221 rows x 15 columns]

In [69]:
df[0:10] #slice by rows

County  License Number Operation Type Establishment Type  \
0       ALBANY          750931          Store                 AC   
1        KINGS          619451          Store                 AC   
2        KINGS          751355          Store                 AC   
3     SARATOGA          722816          Store                 AC   
4      SUFFOLK          719057          Store                 AC   
5        BRONX          728045          Store                 AC   
6  WESTCHESTER          752115          Store                 AC   
7      NIAGARA          290011          Store                  A   
8        BRONX          748243          Store                  A   
9        KINGS          730459          Store                 AC   

                      Entity Name                   DBA Name Street Number  \
0                     EVANS JULIA                    TERROSA             8   
1                     159-MP CORP                   FOODTOWN           159   
2                       YDHNY INC                      YDHNY            25   
3                        IANS LLC                       IANS          1500   
4                      HO 110 INC                      EXXON          1350   
5                BIG SAL DELI INC               BIG SAL DELI          1361   
6                   NFM GROUP LLC           NFM GOURMET DELI           231   
7                        ALDI INC                    ALDI 83          6035   
8             PEOPLES DRUG RX INC            PEOPLES DRUG RX          3176   
9  MCKINLEY DELI AND GROCERY CORP  MCKINLEY DELI AND GROCERY           230   

       Street Name Address Line 2 Address Line 3           City State  \
0        VERDA AVE            NaN            NaN    CLARKSVILLE    NY   
1         N 3RD ST            NaN            NaN       BROOKLYN    NY   
2         N 5TH ST            NaN            NaN       BROOKLYN    NY   
3             RT 9            NaN            NaN   CLIFTON PARK    NY   
4           RT 110            NaN            NaN    FARMINGDALE    NY   
5        OGDEN AVE            NaN            NaN          BRONX    NY   
6        NORTH AVE            NaN            NaN   NEW ROCHELLE    NY   
7        PORTER RD            NaN            NaN  NIAGARA FALLS    NY   
8   BAINBRIDGE AVE            NaN            NaN          BRONX    NY   
9  MCKINLEY AVENUE            NaN            NaN       BROOKLYN    NY   

   Zip Code     Area                        Georeference  
0     12041      NaN  POINT (-73.949128787 42.577416307)  
1     11211   9000.0  POINT (-73.959567442 40.715342402)  
2     11249      NaN  POINT (-73.963025857 40.719355977)  
3     12065      NaN  POINT (-73.742465504 42.837221945)  
4     11735   1000.0  POINT (-73.423437971 40.737553896)  
5     10452      NaN  POINT (-73.924907266 40.841683625)  
6     10801      NaN  POINT (-73.781459583 40.910737968)  
7     14305  15000.0   POINT (-78.992660576 43.10655676)  
8     10467      NaN  POINT (-73.879369643 40.875530462)  
9     11208      NaN  POINT (-73.867387964 40.680373689)

#### C. Manipulate the Data

I selected the "City" column in my dataset and sorted the data by the "Area" column. I filtered the data by PKNBGFRESAZXCVBHNJK